src/unix/fs.c

```diff
@@ -231,7 +231,29 @@ static ssize_t uv__fs_futime(uv_fs_t* req) {
 
 
 static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
-  return mkdtemp((char*) req->path) ? 0 : -1;
+  char *path = (char*) req->path;
+
+#if defined(__MVS__)
+  /* There is no mkdtemp. So instead use mktemp to generate a
```

Comment:
I think we can a bit better than having two disparate implementations.  mkdtemp() is basically two operations: 1) generating a random path name, and 2) creating the directory.

1) can be broken out into platform-agnostic and platform-specific parts and 2) can be implemented in terms of uv_fs_mkdir().

The fact that `fs__mkdtemp()` currently operates on wide characters can be sidestepped by not converting to wide input in the Windows version of `uv_fs_mkdtemp()`.

Apropos 1), I suggest something like this:
```c
struct uv__random {
#ifdef _WIN32
  HCRYPTPROV crypto_provider;
#endif
};

int uv__random_init(struct uv__random* r);
int uv__random_generate(struct uv__random* r, void* buf, size_t size);
void uv__random_destroy(struct uv__random* r);
```